### PR TITLE
#261: Implementation of Header Component 

### PR DESCRIPTION
**Pull request by SaqAsh**
## Summary

Implement dictionary header component, based off of the theming as well as adding in tests to exercise edge cases

## Related Issue
- #261 

### UI
- Continued the development of  new component `DictionaryHeader` which displays current dictionary name, version and description
- Added logic to truncate descriptions beyond a character threshold and display a “Show more”/“Read less” toggle
- Add more exemplars inside of the storybook to exercise the truncation state

## Readiness Checklist

- [X] **Self Review**
  - I have performed a self review of code
  - I have run the application locally and manually tested the feature
  - I have checked all updates to correct typos and misspellings
- [X] **Formatting**
  - Code follows the project style guide
  - Autmated code formatters (ie. Prettier) have been run
- [ ] **Local Testing**
  - Successfully built all packages locally
  - Successfully ran all test suites, all unit and integration tests pass
- [ ] **Updated Tests**
  - Unit and integration tests have been added that describe the bug that was fixed or the features that were added
- [ ] Documentation
  - All new environment variables added to `.env.schema` file and documented in the README
  - All changes to server HTTP endpoints have open-api documentation
  - All new functions exported from their module have TSDoc comment documentation

### CODE REVIEW

**Review by joneubank — DISMISSED** [review]

File: packages/ui/src/viewer-table/DictionaryHeader.tsx [hunk 1]
```diff
@@ -20,61 +20,130 @@
...
+	const { ChevronDown } = theme.icons;
+	const [isExpanded, setIsExpanded] = useState(false);
+
+	// Determine if the description is long enough to need a toggle, based off of how many characters we want to show by default
+	// according to the figma styling
+	const needsToggle = description && description.length > DESCRIPTION_LENGTH_THRESHOLD;
+	// We want to show all the text if it is not long or if it is already expanded via state variable
+	const showFull = isExpanded || !needsToggle;
+	// Based off of showFull, we determine the text to show, either its the full description or a truncated version
+	const textToShow = showFull ? description : description.slice(0, DESCRIPTION_LENGTH_THRESHOLD) + '... ';
```
Comment (SaqAsh) [inline packages/ui/src/viewer-table/DictionaryHeader.tsx:125]: Was not too happy with this implementation, I have refactored this into something that appropriately implements this to spec.

**Review by ciaranschutte — DISMISSED** [review]

File: packages/ui/src/viewer-table/DictionaryHeader.tsx [hunk 2]
```diff
@@ -20,61 +20,130 @@
...
-					>
-						{description}
-					</p>
+					<div css={descriptionColumnStyle}>
+						<div>
+							<span css={descriptionStyle(theme)}>{textToShow}</span>
+							{needsToggle && (
+								<span css={linkStyle(theme)} onClick={() => setIsExpanded((prev) => !prev)}>
+									{' '}
+									{isExpanded ? 'Read less' : 'Show more'}
```
[the remark this answers is not in the document]
Comment (SaqAsh) [reply packages/ui/src/viewer-table/DictionaryHeader.tsx:141]: Prettier formatting just forces it to be in the next line for some reason....

**Review by ciaranschutte — APPROVED** [review]
